reactive/kubernetes_worker.py

```diff
@@ -1175,6 +1175,7 @@ def clear_cloud_flags():
     remove_state('kubernetes-worker.cloud.request-sent')
     remove_state('kubernetes-worker.cloud.blocked')
     remove_state('kubernetes-worker.cloud.ready')
+    set_state('kubernetes-worker.restart-needed')  # force restart
```

Comment:
The conditions for this handler:

```
@when_none('endpoint.aws.joined',
           'endpoint.gcp.joined',
           'endpoint.openstack.joined',
           'endpoint.vsphere.joined',
           'endpoint.azure.joined')
```

When kubernetes-worker isn't related to any cloud providers, this condition will always be true. As a consequence, every hook will set `kubernetes-worker.restart-needed`.

Reply:
Yeah, I caught this yesterday and fixed it on the master, but forgot the worker. Fixing now.